ksp_plugin/plugin.cpp

```diff
@@ -666,29 +666,49 @@ Velocity<World> Plugin::BubbleVelocityCorrection(
                                                  reference_body));
 }
 
-FrameField<World> Plugin::Navball(
+std::unique_ptr<FrameField<World, Navball>> Plugin::NavballFrameField(
     Position<World> const& sun_world_position) const {
-  auto const to_world =
-      OrthogonalMap<WorldSun, World>::Identity() * BarycentricToWorldSun();
-  ephemeris_->Prolong(current_time_);
-  auto const positions_from_world =
-      AffineMap<World, Barycentric, Length, OrthogonalMap>(
-          sun_world_position,
-          sun_->current_position(current_time_),
-          to_world.Inverse());
-  return [this, positions_from_world, to_world](
-      Position<World> const& q) -> Rotation<World, World> {
-    // KSP's navball has x west, y up, z south.
-    // we want x north, y west, z up.
-    auto const orthogonal_map = to_world *
-        plotting_frame_->FromThisFrameAtTime(current_time_).orthogonal_map() *
-        Permutation<World, Navigation>(
-            Permutation<World, Navigation>::XZY).Forget() *
-        Rotation<World, World>(π / 2 * Radian,
-                               Bivector<double, World>({0, 1, 0})).Forget();
-    CHECK(orthogonal_map.Determinant().Positive());
-    return orthogonal_map.rotation();
+
+  class NavballFrameField : public FrameField<World, Navball> {
```

Comment:
I don't like this class in its current state; what's actually being returned is `CoordinateFrame<Navigation, RightHandedNavball>`, whose coordinate (lack of) rotation is composed left and right with (position-independent) orthogonal maps. Here the dependence on `CoordinateFrame` not made explicit, in fact `RightHandedNavball` is never formalized. This matters, because when `FrameField`s other than `CoordinateField` are used (that's the whole point of `BodySurfaceFrameField`), we can just put them through the same thing.
Thus, this class should take a `CoordinateFrame<Navigation, RightHandedNavball>` at construction, and use that. `RightHandedNavball` need not be exposed too high; a forward-declaration within `NavballFrameField` will do the job well enough.